crates/host-utils/src/fetcher.rs

```diff
@@ -72,6 +73,63 @@ impl SP1KonaDataFetcher {
         }
     }
 
+    /// Get the earliest L1 header in a batch of boot infos.
+    pub async fn get_earliest_l1_head_in_batch(
+        &self,
+        boot_infos: &[RawBootInfo],
+    ) -> Result<Header> {
+        let mut earliest_block_num: u64 = u64::MAX;
+        let mut earliest_l1_header: Option<Header> = None;
+
+        for boot_info in boot_infos {
+            let l1_block_header = self
+                .get_header_by_hash(ChainMode::L1, boot_info.l1_head)
+                .await?;
+            if l1_block_header.number < earliest_block_num {
+                earliest_block_num = l1_block_header.number;
+                earliest_l1_header = Some(l1_block_header);
+            }
+        }
+        Ok(earliest_l1_header.unwrap())
+    }
+
+    /// Fetch headers for a range of blocks inclusive.
+    pub async fn fetch_headers_in_range(&self, start: u64, end: u64) -> Result<Vec<Header>> {
```

Comment:
Make sure to `dotenv::dotenv().ok();` here. This was failing for me before because with that missing it was trying to use localhost for L1 RPC.

Reply:
Added it when init'ing `SP1KonaDataFetcher`